app/models/manageiq/providers/ibm_power_hmc/infra_manager/event_target_parser.rb

```diff
@@ -20,35 +20,37 @@ def parse
     case ems_event.event_type
     when "MODIFY_URI", "ADD_URI", "DELETE_URI" # Damien: INVALID_URI?
       $ibm_power_hmc_log.info("#{self.class}##{__method__} #{ems_event.event_type} #{ems_event.full_data}")
-      elems = elem(raw_event[:data])
-      case elems[:type]
+      elems = raw_event[:data]
+        .match(%r{^.+/rest/api/uom/(?:ManagedSystem/(?<manager_uuid>.+)/)?(?<type>.+)/(?<uuid>.+)$})
```

Comment:
Some more more regex suggestions

- Prefer a more specific matcher than `.+`, such as `[^/]+` (assuming you want to stop at the next `/` character, this is better and won't create weird backtracking)
- Remove the starting `^.+` bit because it's not useful and you don't actually need it.  As long as you have the tail anchor, it's the same result, but more performant.